src/vs/workbench/contrib/tasks/common/taskConfiguration.ts

```diff
@@ -1606,6 +1606,9 @@ namespace CustomTask {
 		if (CommandConfiguration.hasCommand(task.command) || task.configurationProperties.dependsOn === undefined) {
 			task.command = CommandConfiguration.fillGlobals(task.command, globals.command, task.configurationProperties.name);
 		}
+		if (task.configurationProperties.name) {
+			task.command.name = task.configurationProperties.name;
```

Comment:
I don't know if this is safe to do, but it doesn't feel like it belongs in `fillGlobals`, as `fillGlobals` is about coping global task properties into an individual task.